File: test/fixtures/java-unirest/post_binary_file.java
import kong.unirest.HttpResponse;
import kong.unirest.Unirest;

class Main {
    public static void main(String[] args) {
            HttpResponse<String> httpResponse
               = Unirest.post("http://localhost:28139/american-art/query")
            .header("Content-type", "application/sparql-query")
            .header("Accept", "application/sparql-results+json")
            .asString();
    }
}

File: test/fixtures/ruby-httparty/post_binary_file.rb
require 'httparty'

url = 'http://localhost:28139/american-art/query'
headers = {
  'Content-type': 'application/sparql-query',
  'Accept': 'application/sparql-results+json',
}
body = File.binread('./sample.sparql').delete("\n")
res = HTTParty.post(url, headers: headers, body_stream: body)
